### DISCUSSION TITLE
issue when compile with pyinstaller

### DISCUSSION THREAD

**Discussion by xalteropsx** [fastapi/fastapi#11623] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
uvicorn.run(app ="main:app"
```

### Description

no idea what going on
![image](https://github.com/tiangolo/fastapi/assets/103671642/ae01ee42-9a87-4d26-8037-6a2074613228)

### Operating System

Windows

### Operating System Details

window 11

### FastAPI Version

3.11

### Pydantic Version

2.0

### Python Version

3.11

### Additional Context

sad

**Comment by PhysicallyActive** [comment]
This looks more like a setup problem than a `fastapi` problem. Can you give us a [MRE](https://en.wikipedia.org/wiki/Minimal_reproducible_example) of your project, as well as how you execute it?

**Reply by xalteropsx** [reply]
```py
if __name__ == "__main__":
    import uvicorn
    uvicorn.run(app="main:app", host="127.0.0.1", port=5373)
```

**Reply by PhysicallyActive** [reply, marked as answer]
I'm sorry, but this is not a MRE (because I can't execute it).
How does your tree structure look? Like this?

```
my-project/
    main.py
    pyproject.toml
```

Does your `main.py` file look like this (or some other way)?

```python
# main.py

from fastapi import FastAPI

app = FastAPI()

@app.get("/")
async def hello_world() -> str:
    return "Hello, World!"

if __name__ == "__main__":
    import uvicorn
    uvicorn.run(app="main:app", host="127.0.0.1", port=5373)
```

Do you try to execute it as `python main.py`?

**Reply by xalteropsx** [reply]
yah i do like that i using it seems like it cant run on multiple executable file rather run on single file which is pretty slower anyway for now i will close this issue if i cant found out i will reproduce with minimal code so u can play around with it as well between thnx alot bro
